File: 0-实验/0-实验报告/python/根轨迹.py
import numpy as np
import matplotlib.pyplot as plt
import control as ctrl


# 定义函数绘制根轨迹
def plot_root_locus(a):
    # 定义传递函数
    num = [1, 1]  # 对应 K(s + 1)
    den = [1, 0, a]  # 对应 s^2(s + a)
    system = ctrl.TransferFunction(num, den)

    # 计算根轨迹并绘制
    rlist, klist = ctrl.root_locus(system, plot=True)
    plt.title(f"根轨迹图 (a = {a})")
    plt.xlabel("实部")
    plt.ylabel("虚部")
    plt.grid()
    plt.show()


# 绘制不同 a 值的根轨迹
for a in [8, 9, 10]:
    plot_root_locus(a)

File: 0-实验报告/mworks/根轨迹.jl
using ControlSystems
using TyPlot

# 定义绘制根轨迹的函数
function plot_root_locus(a)
    # 定义传递函数 G(s)
    num = [1, 1]      # 对应 K(s + 1)
    den = [1, 0, a]   # 对应 s^2(s + a)
    sys = tf(num, den)

    # 绘制根轨迹
    rlocus(sys)
    title!("根轨迹图 (a = $a)")
    xlabel!("实部")
    ylabel!("虚部")
    grid!(true)
end

# 绘制不同 a 值的根轨迹
for a in [8, 9, 10]
    plot_root_locus(a)
end
